fuel-tx/src/tests/valid_cases/transaction.rs

```diff
@@ -395,6 +395,111 @@ fn create__check__no_max_fee_fails() {
     assert_eq!(ValidityError::TransactionMaxFeeNotSet, err);
 }
 
+#[test]
+fn script__check__not_set_owner_success() {
+    // Given
+    let rng = &mut StdRng::seed_from_u64(8586);
+    let block_height = 1000.into();
+
+    // When
+    let result = TransactionBuilder::script(generate_bytes(rng), generate_bytes(rng))
+        .add_fee_input()
+        .finalize()
+        .check(block_height, &test_params());
+
+    // Then
+    assert!(result.is_ok());
+}
+
+#[test]
+fn script__check__set_owner_success() {
+    // Given
+    let rng = &mut StdRng::seed_from_u64(8586);
+    let block_height = 1000.into();
+
+    // When
+    let result = TransactionBuilder::script(generate_bytes(rng), generate_bytes(rng))
+        .add_fee_input()
+        .owner(0)
+        .finalize()
+        .check(block_height, &test_params());
+
+    // Then
+    assert!(result.is_ok());
+}
+
+#[test]
+fn script__check__set_owner_bad_idx() {
+    // Given
+    let rng = &mut StdRng::seed_from_u64(8586);
+    let block_height = 1000.into();
+
+    // When
+    let err = TransactionBuilder::script(generate_bytes(rng), generate_bytes(rng))
+        .add_fee_input()
+        .owner(1)
+        .finalize()
+        .check(block_height, &test_params())
+        .expect_err("Expected erroneous transaction");
+
+    // Then
+    assert_eq!(ValidityError::TransactionOwnerIndexDoesntExist, err);
+}
```

Comment:
following on this test, I suggest to add another one: 

```rust
fn script__check__set_owner_bad_input_type() {
    // Given
    let rng = &mut StdRng::seed_from_u64(8586);
    let block_height = 1000.into();

    // When
    let err = TransactionBuilder::script(generate_bytes(rng), generate_bytes(rng))
        .add_fee_input()
        .add_contract_input()
        .owner(1)
        .finalize()
        .check(block_height, &test_params())
        .expect_err("Expected erroneous transaction");

    // Then
    assert_eq!(ValidityError::TransactionOwnerInputHasNoOwner, err);
}
```

It requires you to add a method `TransactionBuilder::add_contract_input()`, but this can be easily done: 

```rust
    pub fn add_contract_input(&mut self) -> &mut Self {
        let contract_input = Input::Contract(Contract::default());
        self.tx.add_input(contract_input);
        self
    }
```

(Btw, if you want to be pedantic you could also test with other inputs, and rewrite `add_contract_input` to take a `SeedableRng` in input).

Reply:
I added your test and reuse some code already existing to create the contract because adding this in the public interface used by all Rust project inside a PR that has no meanign with that, I'm not confident.